electron/main/database/TableHelperFunctions.ts

```diff
@@ -101,44 +101,62 @@ const getTableAsArray = async (
   return results;
 };
 
-const computeDbItemsToAdd = async (
+const computeDbItemsToAddOrUpdate = async (
   filesInfoList: FileInfo[],
   tableArray: DBEntry[]
 ): Promise<DBEntry[][]> => {
-  const promises = filesInfoList.map(convertFileTypeToDBType);
-
-  const filesAsChunksToAddToDB = await Promise.all(promises);
+  const filesAsChunks = await convertFileInfoListToDBItems(filesInfoList);
 
-  return filesAsChunksToAddToDB.filter((chunksBelongingToFile) =>
-    filterChunksNotInTable(chunksBelongingToFile, tableArray)
+  const fileChunksMissingFromTable = filesAsChunks.filter(
+    (chunksBelongingToFile) =>
+      areChunksMissingFromTable(chunksBelongingToFile, tableArray)
   );
+
+  return fileChunksMissingFromTable;
+};
+
+const convertFileInfoListToDBItems = async (
+  filesInfoList: FileInfo[]
+): Promise<DBEntry[][]> => {
+  const promises = filesInfoList.map(convertFileTypeToDBType);
+  const filesAsChunksToAddToDB = await Promise.all(promises);
+  return filesAsChunksToAddToDB;
 };
 
 const computeDBItemsToRemoveFromTable = async (
   filesInfoList: FileInfo[],
   tableArray: DBEntry[]
 ): Promise<DBEntry[]> => {
-  const notInFilesInfoList = tableArray.filter(
+  const itemsInTableAndNotInFilesInfoList = tableArray.filter(
     (item) => !filesInfoList.some((file) => file.path == item.notepath)
   );
-  return notInFilesInfoList;
+  return itemsInTableAndNotInFilesInfoList;
 };
 
-const filterChunksNotInTable = (
-  chunksBelongingToFile: DBEntry[],
+const areChunksMissingFromTable = (
+  chunksToCheck: DBEntry[],
   tableArray: DBEntry[]
 ): boolean => {
-  if (chunksBelongingToFile.length == 0) {
+  // checking whether th
+  if (chunksToCheck.length == 0) {
+    // if there are no chunks and we are checking whether the table
     return false;
   }
-  if (chunksBelongingToFile[0].content == "") {
+
+  if (chunksToCheck[0].content === "") {
     return false;
   }
-  const notepath = chunksBelongingToFile[0].notepath;
+  // then we'd check if the filepaths are not present in the table at all:
+  const notepath = chunksToCheck[0].notepath;
   const itemsAlreadyInTable = tableArray.filter(
     (item) => item.notepath == notepath
   );
-  return chunksBelongingToFile.length != itemsAlreadyInTable.length;
+  if (itemsAlreadyInTable.length == 0) {
```

Comment:
What happens for an empty file with no contents? 

Would it have just a default chunk with some delimiters added?

Reply:
Line 146 would ensure it doesn't get indexed